flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/catalog/DatabaseCalciteSchema.java

```diff
@@ -68,25 +68,28 @@ public Table getTable(String tableName) {
 		return catalogManager.getTable(identifier)
 			.map(result -> {
 				CatalogBaseTable table = result.getTable();
-				Catalog catalog = catalogManager.getCatalog(catalogName).get();
-				FlinkStatistic statistic = getStatistic(result.isTemporary(),
-					catalog, table, identifier);
+				FlinkStatistic statistic = getStatistic(result.isTemporary(), table, identifier);
 				return new CatalogSchemaTable(identifier,
 					table,
 					statistic,
-					catalog.getTableFactory().orElse(null),
+					catalogManager.getCatalog(catalogName)
+						.flatMap(Catalog::getTableFactory)
```

Comment:
Did you mean a temporal object could exist without a catalog ? For example, the view ?

Reply:
Not a temporal, but temporary. Yes, tables, views, functions, etc. That's what we discussed and agreed upon in FLIP-64 and FLIP-57.